test/test_helper.rb

```diff
@@ -24,6 +28,58 @@ def stub_remote_file(url, filename)
     stub_content(url, body: File.read(Rails.root.join('test', 'support', filename)))
   end
 
+  # Stubs a search for a term and a project with random data
+  # opts :matches sets the number of results, otherwise random
+  def stub_search_random(term, baseproject, opts = {})
+    xpath = "contains-ic(@name,%20'#{term}')%20and%20path/project='#{baseproject}'%20and%20not(contains-ic(@name,%20'-debuginfo'))%20and%20not(contains-ic(@name,%20'-debugsource'))%20and%20not(contains-ic(@name,%20'-devel'))%20and%20not(contains-ic(@name,%20'-lang'))"
+
+    matches = (opts[:matches] || (1..10).to_a.sample).to_i
+
+    builder = Nokogiri::XML::Builder.new do |xml|
+      xml.collection(matches: matches) do
+        matches.times.each do
+          pkg = term
+          bin = "#{pkg}-#{Faker::Lorem.unique.word}"
```

Comment:
why do you need the word to be unique? :thinking:

Reply:
Because the same package can not produce two sub-packages with the same name, and there is no facility to make it unique at the package level only.